### PR TITLE
readability improvement for metadata class

### PR DESCRIPTION
**Pull request by Renkai**

### CODE REVIEW

File: cpp/src/lance/format/metadata.cc [hunk 1]
```diff
@@ -68,16 +68,19 @@ int32_t Metadata::GetBatchLength(int32_t batch_id) const {
...
   if (row_index < 0 || row_index >= len) {
-    return ::arrow::Status::IndexError(fmt::format("Row index out of range: {} of {}", row_index, len));
+    return ::arrow::Status::IndexError(fmt::format("Row index out of range: {} of {}", row_index, len - 1));
   }
   auto it = std::upper_bound(pb_.batch_offsets().begin(), pb_.batch_offsets().end(), row_index);
   if (it == pb_.batch_offsets().end()) {
     return ::arrow::Status::IndexError("Row index out of range {} of {}", row_index, len);
   }
-  int32_t bound_idx = std::distance(pb_.batch_offsets().begin(), it);
+  int32_t bound_idx = std::distance(pb_.batch_offsets().begin(), it) - 1;
```
Comment (eddyxu) [inline cpp/src/lance/format/metadata.cc:82]: what if the `it == batch_offset().begin()` ? in such case bound_idx = -1?

Comment (Renkai) [reply cpp/src/lance/format/metadata.cc:82]: The first value of offsets is always 0, `row_index` is always not less than 0, so this case should not happen unless we change how offsets works.
https://github.com/eto-ai/lance/blob/861230d0d8186312e18f206fb8262eb919828549/cpp/src/lance/format/metadata.cc#L59

Comment (eddyxu) [reply cpp/src/lance/format/metadata.cc:82]: ok, got it.  It seems line #83 `assert` is always true as well? lets remove it?

File: cpp/src/lance/format/metadata.cc [hunk 2]
```diff
@@ -68,16 +68,19 @@ int32_t Metadata::GetBatchLength(int32_t batch_id) const {
 
 ::arrow::Result<std::tuple<int32_t, int32_t>> Metadata::LocateBatch(int32_t row_index) const {
   int64_t len = length();
+  if (len == 0) {
+    return ::arrow::Status::IndexError(fmt::format("The offsets table is empty"));
```
Comment (eddyxu) [inline cpp/src/lance/format/metadata.cc:72]: don't need `fmt::format` here

File: cpp/src/lance/format/metadata.cc (hunk 1 above)
Comment (Renkai) [reply cpp/src/lance/format/metadata.cc:82]: updated

**Review by eddyxu — APPROVED** [review]

**Pull request merged**